src/Elasticsearch/Framework/ElasticsearchHelper.php

```diff
@@ -63,6 +63,8 @@ public function __construct(
 
     public function logOrThrowException(\Throwable $exception): bool
```

Comment:
now the method name does not say anymore what really happens in here :grin:

Reply:
@mitelg that was my previous comment about https://github.com/shopware/platform/pull/2035#issuecomment-920739272 so shall we change it as well?

Reply:
oh, I did not read that :see_no_evil: 

renaming the method would be a breaking change. what is possible, is to add a new method and deprecate this one which then only calls the new one

Reply:
Yes sure. I would follow the deprecation guide :) I can do that

Reply:
I hate myself already for this as I decorated it and it probably would break sth in my decoration... :(

Reply:
maybe we need to introduce an abstract class for that? :thinking: @keulinho @shyim @OliverSkroblin what do you think?

Reply:
Ok you'll figure this one out @mitelg @JoshuaBehrens. Just hit me up when this is ready to be imported

Reply:
@mitelg do you think about to create a Decoratable?

Reply:
I am fine with an abstract class.

Reply:
So just abstract class? What is not abstract on it? When it is an abstract class, shall I make it a decoratable? Otherwise an interface would be sufficient right?

Reply:
we don't make interfaces anymore and use abstract classes for decoration pattern. see https://github.com/shopware/platform/blob/trunk/src/Core/Content/Seo/AbstractSeoResolver.php#L5 as example

Reply:
That was what I tried to find out. I make it a decoratable then